launchable/commands/record/tests.py

```diff
@@ -140,47 +144,75 @@ def payload():
-                                    yield ','
-                                first = False
-                                count += 1
+                                yield json.dumps(CaseEvent.from_case_and_suite(self.path_builder, case, suite, report))
 
-                                yield json.dumps(CaseEvent.from_case_and_suite(self.path_builder, case, suite, p))
                     except Exception as e:
-                        raise Exception("Failed to process a report file: %s" % p) from e
+                        exceptions.append(Exception(
+                            "Failed to process a report file: {}".format(report), e))
+
+                if len(exceptions) > 0:
+                    # defer XML persing exceptions
+                    raise Exception(exceptions)
+
+            def splitter(iterable: Generator, size: int) -> Generator[Generator, None, None]:
+                return ichunked(iterable, size)
+
+            # generator that creates the payload incrementally
+            def payload(cases: Generator[TestCase, None, None]) -> Generator[str, None, None]:
+                nonlocal count
+                yield '{"events": ['
+                first = True        # used to control ',' in printing
+                for case in cases:
+                    if not first:
+                        yield ','
+                    first = False
+                    count += 1
+                    yield case
                 yield ']}'
 
-            def printer(f):
+            def printer(f: Generator) -> Generator:
                 for d in f:
                     print(d)
                     yield d
 
-            payload = printer(payload()) if debug else payload()
-
-            # str -> bytes then gzip compress
-            payload = (s.encode() for s in payload)
-            payload = compress(payload)
+            def send(payload: Generator[TestCase, None, None]) -> None:
+                # str -> bytes then gzip compress
+                payload = (s.encode() for s in payload)
+                payload = compress(payload)
 
-            headers = {
-                "Content-Type": "application/json",
-                "Content-Encoding": "gzip",
-            }
-
-            client = LaunchableClient(token)
-
-            try:
+                headers = {
+                    "Content-Type": "application/json",
+                    "Content-Encoding": "gzip",
+                }
+                client = LaunchableClient(token)
                 res = client.request(
                     "post", "{}/events".format(session_id), data=payload, headers=headers)
                 res.raise_for_status()
-                res = client.request("patch", "{}/close".format(session_id), headers=headers)
+
+            try:
+                splitted_cases = splitter(printer(testcases(
+                    self.reports)), CASE_POST_CHUNK) if debug else splitter(testcases(self.reports), CASE_POST_CHUNK)
```

Comment:
Nit: at least `splitter(printer(testcases(self.reports)) if debug else testcases(self.reports), CASE_POST_CHUNK)` for a strength reduction. Beyond that is subjective, but to me the following code is more readable:
```
tc = testcases(self.reports)
if debug: tc=printer(tc)
for chunk in splitter(tc):
    ...
```